### PR TITLE
Make kitchen package work

### PR DESCRIPTION
**Pull request by ccope**
Tested with the following config:
```
---
driver:
  name: vagrant
  provider: lxc

provisioner:
  name: chef_zero

platforms:
  - name: xenial64-lxc
    driver_plugin: vagrant
    driver_config:
      box: hrishikeshdate/xenial64
      ssh:
        insert_key: false

suites:
  - name: mysuite
    run_list:
      - recipe[mycookbook::default]
```

### CODE REVIEW

**Comment by cheeseplus** [comment]
Looks like there are still some failures with the linting from the Travis build - also would love to get testing for the newly introduced command as well.

**Comment by ccope** [comment]
Fixed the lint errors and squashed

File: lib/kitchen/driver/vagrant.rb [hunk 1]
```diff
@@ -135,6 +135,20 @@ def destroy(state)
...
+          raise UserError, "Vagrant instance not created!"
+        end
+        if not (config[:ssh] && config[:ssh][:insert_key] == false)
+          m = "Disable vagrant ssh key replacement to preserve the default key!"
+          warn(m)
+        end
+        instance.transport.connection(state).close
+        box_name = File.join(Dir.pwd, instance.name + ".box")
+        run("#{config[:vagrant_binary]} package --output #{box_name}")
+        destroy(state)
```
Comment (mwrock) [inline lib/kitchen/driver/vagrant.rb]: Is it necessary/desirable to destroy?

Comment (ccope) [inline lib/kitchen/driver/vagrant.rb]: The package command stops the machine, and I found testkitchen had trouble communicating with it if I re-ran vagrant up (ssh would keep failing). I'm not sure why.

Comment (mwrock) [inline lib/kitchen/driver/vagrant.rb]: oh ok. Perhaps vagrant changes the state of the box. That's fair.

**Comment by n-rodriguez** [comment]
Hi! Would it be possible to customize the output filename in `.kitchen.yml`?

**Comment by n-rodriguez** [comment]
Hi there! Any news?

**Comment by ccope** [comment]
I don't currently have the time to add more to this PR, but it seems like it would be a relatively easy change to make after it is merged.

**Comment by cheeseplus** [comment]
Sorry this one was overlooked for so long, I thought we still had some questions to answer but this is safe enough and we can extend later.

**Pull request merged**